crates/trie/sparse/src/trie.rs

```diff
@@ -574,10 +574,18 @@ impl<P> RevealedSparseTrie<P> {
         }
     }
 
-    /// Returns a list of paths to the nodes that were changed according to the prefix set and are
-    /// located at the provided depth when counting from the root node. If there's a leaf at a
-    /// depth less than the provided depth, it will be included in the result.
-    fn get_changed_nodes_at_depth(&self, prefix_set: &mut PrefixSet, depth: usize) -> Vec<Nibbles> {
+    /// Returns a list of levels and paths to the nodes that were changed according to the prefix
+    /// set and are located at the provided depth when counting from the root node. If there's a
+    /// leaf at a depth less than the provided depth, it will be included in the result.
+    ///
+    /// Additionally, returns a new prefix set containing the paths that will not be updated, thus
+    /// need re-calculation.
+    fn get_changed_nodes_at_depth(
+        &self,
+        prefix_set: &mut PrefixSet,
+        depth: usize,
+    ) -> (Vec<(usize, Nibbles)>, PrefixSetMut) {
```

Comment:
with these changes the type returned doesn't match the method name. It is also quite complex, looking at the type itself it is not obvious for instance what the usize is, wdyt?